### DISCUSSION TITLE
Edgeout color

### DISCUSSION THREAD

**Discussion by bioworldmerch** [ImageMagick/ImageMagick#5006] [Digital Image Processing]
Im running  ImageMagick 7.1.0-25 on Mac and trying to solve an issue of taking nearest color and expand it outwards into black or a specified color in a fast way.

My attempt was to take the original image,
1. replace black with transparency - convert input.png -fuzz 30% -transparent black transparent.png
2. morph edge - convert transparent.png -morphology edge diamond:10 output.png

it seems to move the original image edge out 5 and in 5. Not sure I completely understand what it is doing.
Anyone have any ideas how I can expend the nearest color centered on the transparent edge without moving the edge of the original? It has intermediate colors as it got closer to the black original edge that I am looking to cover with the inside color.

One idea I thought is if I could take the outside edge and contract it as specified amount, it would help but can't figure out how to do this.

**Comment by snibgo** [comment]
I don't understand what you want to do. Sample input and output images would help.

If there is only one colour ("MYCOL") you want to "expand outward" to replace black, perhaps "-fill MYCOL -opaque Black".

**Reply by bioworldmerch** [reply]
Thanks for the followup. Im trying to "spread" the nearest colors into the black making it smaller. Basically I want to remove the blended colors between Black and another color touching having Anti-Aliasing for a hard pixelated aliased edge. Im thinking if I could replace the black with transparency which I know how to do. Then I would need to:

Expand the transparency a specified amount like 3-5 pixels smaller.
Then I could EdgeOut the image 6-10 pixels or spread the color out.

Im not sure how to do either.

**Reply by bioworldmerch** [reply]
I would then threshold the black and put on top of the image with spread colors and it would be an Aliased edge pixelated the way I need it.

**Comment by fmw42** [comment]
Did you try -morphology edgeout?

**Reply by bioworldmerch** [reply]
Good call. Testing now. Not sure if I have the commands correct. I thought I had to do some sort of -sparse-color and was trying to follow this link https://legacy.imagemagick.org/discourse-server/viewtopic.php?p=138971#p138971 but I think this will be faster if works out.

Steps I'm testing:
1. replace black with transparency - convert input.png -fuzz 30% -transparent black transparent.png
2. trim edge of transparency inward - convert transparent.png -channel A  -morphology Erode Diamond:3 +channel transparent.png
3. morph edge out 6 pixels - convert transparent.png -morphology edgeout diamond:5

**Comment by snibgo** [comment]
@bioworldmerch: to help us understand, please show sample input and output images.
